File: app-email-notifier/src/main/java/com/github/arhor/aws/microservices/playground/notifications/repository/UserRepository.java
package com.github.arhor.aws.microservices.playground.notifications.repository;

import com.github.arhor.aws.microservices.playground.notifications.model.User;

import java.util.Optional;

public interface UserRepository {

    Optional<User> findById(long userId);

    void save(User user);

    void deleteById(long userId);
}

File: app-users/src/main/kotlin/com/github/arhor/aws/microservices/playground/users/data/repository/UserRepository.kt
package com.github.arhor.aws.microservices.playground.users.data.repository

import com.github.arhor.aws.microservices.playground.users.data.model.User
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface UserRepository : CoroutineCrudRepository<User, Long> {

    suspend fun existsByEmail(email: String): Boolean
}
